packages/api/tests/endpoints/test-granules.js

```diff
@@ -1976,6 +1978,113 @@ test.serial('PUT replaces an existing granule in all data stores', async (t) =>
   );
 });
 
+test.serial('PUT creates a granule if one does not already exist in all data stores', async (t) => {
+  const {
+    knex,
+  } = t.context;
+
+  const granuleId = `${cryptoRandomString({ length: 7 })}.${cryptoRandomString({ length: 20 })}.hdf`;
+
+  const fakeGranule = fakeGranuleFactoryV2({
+    granuleId,
+    status: 'completed',
+    execution: t.context.executionUrl,
+    duration: 47.125,
+  });
+
+  await request(app)
+    .put(`/granules/${fakeGranule.granuleId}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(fakeGranule)
+    .expect(201);
+
+  const fakePgGranule = await translateApiGranuleToPostgresGranule({
+    dynamoRecord: fakeGranule,
+    knexOrTransaction: knex,
+  });
+
+  const actualGranule = await t.context.granuleModel.get({
+    granuleId: fakeGranule.granuleId,
+  });
+
+  t.deepEqual(actualGranule, {
+    ...fakeGranule,
+    error: {},
+    timestamp: actualGranule.timestamp,
+  });
+
+  const actualPgGranule = await t.context.granulePgModel.get(knex, {
+    granule_id: actualGranule.granuleId,
+    collection_cumulus_id: t.context.collectionCumulusId,
+  });
+
+  t.deepEqual(
+    omitBy(actualPgGranule, (x) => x === null),
```

Comment:
Worth having a helper that only removes `nulls` instead of null + undefined?

Reply:
Also maybe we should use an explicit `omitList` here?

Reply:
I think in this instance it's just a function of using knex.get that undefineds come back as null.    Given that, I'm opting to go with: bb20d73e5b .  Let me know if that works.